crates/fluvio-smartmodule-derive/src/ast.rs

```diff
@@ -30,30 +34,27 @@ pub enum SmartModuleKind {
 }
 
 impl SmartModuleKind {
-    #[allow(clippy::ptr_arg)]
-    fn from_ast(args: &AttributeArgs) -> SynResult<Self> {
-        let ss_type =
-            args.iter()
-                .find_map(|it| match it {
-                    NestedMeta::Meta(Meta::Path(it)) => {
-                        it.segments.iter().rev().next().and_then(|it| {
-                            match &*it.ident.to_string() {
-                                "aggregate" => Some(Self::Aggregate),
-                                "filter" => Some(Self::Filter),
-                                "map" => Some(Self::Map),
-                                "array_map" => Some(Self::ArrayMap),
-                                "filter_map" => Some(Self::FilterMap),
-                                "init" => Some(Self::Init),
-                                "look_back" => Some(Self::LookBack),
-                                _ => None,
-                            }
-                        })
-                    }
-                    _ => None,
-                })
-                .ok_or_else(|| SynError::new(args[0].span(), "Missing SmartModule type"))?;
+    fn parse(meta: ParseNestedMeta) -> SynResult<Option<Self>> {
+        let maybee_ss_type = match &*meta
+            .path
+            .get_ident()
+            .ok_or_else(|| SynError::new(meta.path.span(), "Missing SmartModule type"))?
+            .to_string()
+        {
+            "aggregate" => Some(Self::Aggregate),
+            "filter" => Some(Self::Filter),
+            "map" => Some(Self::Map),
+            "array_map" => Some(Self::ArrayMap),
+            "filter_map" => Some(Self::FilterMap),
+            "init" => Some(Self::Init),
+            "look_back" => Some(Self::LookBack),
+            _ => None,
+        };
+
+        let ss_type = maybee_ss_type
+            .ok_or_else(|| SynError::new(meta.path.span(), "Invalid SmartModule type"))?;
 
-        Ok(ss_type)
+        Ok(Some(ss_type))
```

Comment:
If this function is capable to return always value, why do we make it optionable and then deal with option instead of value?